lib/IRGen/GenType.cpp

```diff
@@ -423,6 +423,25 @@ static llvm::Value *computeExtraTagBytes(IRGenFunction &IGF, IRBuilder &Builder,
   return phi;
 }
 
+static Address alignAddress(IRGenFunction &IGF, Address originAddress,
+			    llvm::Value *shift) {
+  auto &Builder = IGF.Builder;
+  auto &IGM = IGF.IGM;
+  auto *alignedAddressValue =
+      Builder.CreateBitCast(originAddress, IGM.Int8PtrTy).getAddress();
+  Address alignedAddress = Builder.CreateBitCast(originAddress, IGM.Int8PtrTy);
+  if (IGM.Triple.getOS() == llvm::Triple::Linux &&
```

Comment:
I think it would be better to just make this check an assertion if you are worried that it doesn't work on other OSes.

```
assert(IGM.Triple.getOS() == llvm::Triple::Linux && "untested in non-Linux OSes");
```

Reply:
@compnerd if using an `assert`, the program will automatically terminate when the condition is not met, in -Onone debug build. However, that is not what we need. By using `if` statement, it only skips the code in the braces when the condition is not met. Everything else should work normally.

Reply:
It seems extremely unlikely that this would be Linux-specific in any way. I would prefer we just remove the condition.

Reply:
@jckarter will do.